hazelcast/src/main/java/com/hazelcast/internal/util/executor/PoolExecutorUnblockableThreadFactory.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.util.executor;
+
+import com.hazelcast.util.executor.PoolExecutorThreadFactory;
+
+public class PoolExecutorUnblockableThreadFactory extends PoolExecutorThreadFactory {
```

Comment:
minor: can you add a few lines of doc to explain what this thread factory is for?

Reply:
done

Reply:
Can you rename it to UnblockablePoolExecutorThreadFactory?